#!/usr/bin/env python
PACKAGE = "dynamic_arm_pid"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#gen.add(name , type, level, description, default, min, max)
gen.add("arm1_p_gain_pose",    double_t,    0, "arm1_p_gain_pose", 1.3,  0, 2)
gen.add("arm1_i_gain_pose",    double_t,    0, "arm1_i_gain_pose", 0.07,  0, 0.5)
gen.add("arm1_d_gain_pose",    double_t,    0, "arm1_d_gain_pose", 0.2,  0, 0.5)
gen.add("arm1_max_output_pose",    double_t,    0, "arm1_max_output_pose", 50,  0, 50)
gen.add("arm1_p_gain_vel",    double_t,    0, "arm1_p_gain_vel", 0.005,  0, 0.01)
gen.add("arm1_i_gain_vel",    double_t,    0, "arm1_i_gain_vel", 0.00166,  0, 0.002)
gen.add("arm1_d_gain_vel",    double_t,    0, "arm1_d_gain_vel", 0.02,  0, 0.04)
gen.add("arm1_max_output_vel",    double_t,    0, "arm1_max_output_vel", 50,  0, 50)

gen.add("arm2_p_gain_pose",    double_t,    0, "arm2_p_gain_pose", 3.7,  0, 10)
gen.add("arm2_i_gain_pose",    double_t,    0, "arm2_i_gain_pose", 0.3,  0, 0.5)
gen.add("arm2_d_gain_pose",    double_t,    0, "arm2_d_gain_pose", 0.035,  0, 0.04)
gen.add("arm2_max_output_pose",    double_t,    0, "arm2_max_output_pose", 10,  0, 10)
gen.add("arm2_p_gain_vel",    double_t,    0, "arm2_p_gain_vel", 0.0396,  0.01, 1) #0.056
gen.add("arm2_i_gain_vel",    double_t,    0, "arm2_i_gain_vel", 0.009,  0, 0.1) #0.009
gen.add("arm2_d_gain_vel",    double_t,    0, "arm2_d_gain_vel", 0.04,  0, 0.1) #0.04
gen.add("arm2_max_output_vel",    double_t,    0, "arm2_max_output_vel", 10,  0, 10)


exit(gen.generate(PACKAGE, "dynamic_arm_pid", "arm_pid"))
